pyvista/utilities/features.py

```diff
@@ -176,3 +178,67 @@ def transform_vectors_sph_to_cart(theta, phi, r, u, v, w):
     w_t = np.cos(ph) * w - np.sin(ph) * v
 
     return u_t, v_t, w_t
+
+
+def merge(
+        datasets,
+        merge_points=True,
+        main_has_priority=True,
+        progress_bar=False,
+    ):
+    """Merge several datasets.
+
+    .. note::
+       The behavior of this filter varies from the
+       :func:`PolyDataFilters.boolean_union` filter. This filter
+       does not attempt to create a manifold mesh and will include
+       internal surfaces when two meshes overlap.
+
+    datasets : sequence of :class:`pyvista.Dataset`
+        Sequence of datasets. Can be of any :class:`pyvista.Dataset`
+
+    merge_points : bool, optional
+        Merge equivalent points when ``True``. Defaults to ``True``.
+
+    main_has_priority : bool, optional
+        When this parameter is ``True`` and ``merge_points=True``,
```

Comment:
What is the behavior when `main_has_priority=False` and there are more than two datasets. This is also unclear in the overall filter. 

Related. I wonder if it makes more sense in this function to allow the user to choose the index that has priority.

Reply:
I'm pretty sure (but not certain) that it's either the very first or the very last mesh that wins, since the kwarg defines which one to take from pairwise merges.

Reply:
Without looking at the code, I would have guessed the second mesh based on how the filter is setup.  If you aren't certain either, then the documentation is not clear

Reply:
And I didn't look at the documentation, just remembered the implementation :D I'm pretty sure the behaviour is not specified in the docs, yes. "Second mesh [out of n]" would be terrible behaviour, so I'd be inclined to change that if that were the case.

As for the implementation:
https://github.com/pyvista/pyvista/blob/1f86ad262d20a4ef97445c1882b0072410969676/pyvista/core/filters/data_set.py#L4020-L4028
My working assumption has been that whoever is set last as an input will win, which is why `self` is added first if "main does _not_ have priority". And since containers loop forward, the last one would be the last item in the container.